dietpi-install.sh

```diff
@@ -14,6 +16,125 @@ cleanup() {
 # Trap Ctrl+C and other interrupts
 trap cleanup INT TERM
 
+# Verify SHA256 checksum
+verify_sha256() {
+    local image_file="$1"
+    local checksum_url="$2"
+
+    echo "Downloading SHA256 checksum..."
+    if ! wget -q "$checksum_url" -O "${image_file}.sha256"; then
+        echo "Warning: Could not download checksum file from $checksum_url"
+        return 1
+    fi
+
+    echo "Verifying SHA256 checksum..."
+    if ! sha256sum -c "${image_file}.sha256" 2>/dev/null | grep -q "OK"; then
+        echo "ERROR: SHA256 checksum verification FAILED!"
+        echo "The downloaded file may be corrupted or tampered with."
+        return 1
+    fi
+
+    echo "✓ SHA256 checksum verified successfully"
+    return 0
+}
+
+# Import DietPi GPG public key (optional - won't fail if import unsuccessful)
+import_dietpi_gpg_key() {
+    # Check if GPG is available
+    if ! command -v gpg &> /dev/null; then
+        return 1
+    fi
+
+    # DietPi GPG key details
+    local key_id="C2C4D1DEF7C96C6EDF3937B2536B2A4A2E72D870"
+    local key_url="https://github.com/MichaIng.gpg"
+
+    # Check if key is already imported
+    if gpg --list-keys "$key_id" &>/dev/null; then
+        echo "✓ DietPi GPG key already imported"
+        return 0
+    fi
+
+    echo "Importing DietPi GPG public key..."
+
+    # Try to download and import key from GitHub
+    if wget -q "$key_url" -O dietpi.gpg && gpg --import dietpi.gpg &>/dev/null; then
```

Comment:
Let's not mute `gpg` error messages if any, for better debugging:
```suggestion
    if wget -q "$key_url" -O dietpi.gpg && gpg --import dietpi.gpg >/dev/null; then
```